### PR TITLE
[pack] Fix keys APIs for functions without function.json

### LINKED ISSUE #7147: Fix keys APIs for IFunctionProvider functions (without "function.json")
The `/keys` endpoints do not seem to work for functions without "function.json". This is because `KeysController` relies on physical `function.json` to figure out if function name is accurate or not. We need to use `IFunctionMetadataManager` for such operations.

### PR DESCRIPTION
**Pull request by ankitkumarr**
### Issue describing the changes in this PR

resolves https://github.com/Azure/azure-functions-host/issues/7147

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

The PR changes the code to rely on `IFunctionMetadataManager` as the source of truth for functions. The fallback logic is to check in the filesystem to avoid any edge cases for backward compatibility.

### CODE REVIEW

**Review by brettsam — DISMISSED** [review]

File: src/WebJobs.Script.WebHost/Controllers/KeysController.cs [hunk 1]
```diff
@@ -285,9 +287,13 @@ internal bool IsBuiltInSystemKeyName(string keyName)
 
         private bool IsFunction(string functionName)
         {
-            string json = null;
+            if (_functionMetadataManager.TryGetFunctionMetadata(functionName, out _))
+            {
+                return true;
+            }
```
Comment (brettsam) [inline src/WebJobs.Script.WebHost/Controllers/KeysController.cs:294]: I'm wondering if you should flip this to have less impact. Try to read the FunctionConfig first, and then, if false, try this as a fallback? 

I don't know what could break from this, but it's definitely a behavior change like this if the function.json is present.

Comment (fabiocav) [reply src/WebJobs.Script.WebHost/Controllers/KeysController.cs:294]: Agreed. The previous behavior will actually attempt to parse the file. Placing this here does have a behavior impact.

Comment (mathewc) [reply src/WebJobs.Script.WebHost/Controllers/KeysController.cs:294]: Admin APIs like this have to work even if the host isn't running (i.e. keys/functions APIs aren't marked RequiresRunningHost). Doesn't the metadata manager require host startup to initialize codeless extension providers? Or is it all independent of the host running. Should verify this.

Comment (ankitkumarr) [reply src/WebJobs.Script.WebHost/Controllers/KeysController.cs:294]: Yeah I will flip this. @mathewc, metadata manager doesn't rely on Host running. When Host starts, it reinitializes the list of functions it has (including codeless ones). But it shouldn't block on anything like that.

Comment (mathewc) [reply src/WebJobs.Script.WebHost/Controllers/KeysController.cs:294]: Ok

**Review by fabiocav — APPROVED** [review]
